Sources/_StringProcessing/Engine/Processor.swift

```diff
@@ -89,6 +89,8 @@ struct Processor {
   // MARK: Metrics, debugging, etc.
   var cycleCount = 0
   var isTracingEnabled: Bool
+  let shouldMeasureMetrics: Bool
```

Comment:
While well-predicted, do we see any impact from checking this bool inside the processor? Worst case we could make it conditionally compile and the benchmark can enable it.

Reply:
<img width="907" alt="image" src="https://user-images.githubusercontent.com/49693777/184702070-6cbd65e5-c4fa-4f4a-9d74-062834ef2e41.png">

Ah... the answer is quite a lot. Removing both the tracing and metric conditions speeds things up by 10-20% across the board. I'll see if I can figure out the conditional compilation

Reply:
After looking around it doesn't seem like swiftPM supports the feature of "define this compilation variable only when being built as a dependency for target X". The only conditions it can enable/disable compilation variables on are platforms and debug/release. Trying to add multiple targets with flags enabled/disabled for `_StringProcessing` ends up with "overlapping source code" 

https://forums.swift.org/t/swiftpm-and-conditional-compilation/36874
https://forums.swift.org/t/compilation-conditions-and-swift-packages/34627/4